src/import/helm.ts

```diff
@@ -0,0 +1,179 @@
+import { spawnSync } from 'child_process';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { Yaml } from 'cdk8s';
+import { CodeMaker } from 'codemaker';
+// we just need the types from json-schema
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { JSONSchema4 } from 'json-schema';
+import { TypeGenerator } from 'json2jsii';
+import * as semver from 'semver';
+import { ImportBase } from './base';
+import { emitHelmHeader, generateHelmConstruct } from './codegen';
+import { ImportSpec } from '../config';
+
+const MAX_HELM_BUFFER = 10 * 1024 * 1024;
+const CHART_SCHEMA = 'values.schema.json';
+const CHART_YAML = 'Chart.yaml';
+
+export class ImportHelm extends ImportBase {
+  public static async fromSpec(importSpec: ImportSpec): Promise<ImportHelm> {
+    const { source } = importSpec;
+    return new ImportHelm(source);
+  }
+
+  public readonly chartName: string;
+  public readonly chartUrl: string;
+  public readonly chartVersion: string;
+  public readonly chartSchemaPath: string | undefined;
+  public readonly chartDependencies: string[] = [];
+  public readonly tmpDir: string;
+
+  private constructor(source: string) {
+    super();
+
+    const [chartUrl, chartName, chartVersion] = getHelmChartDetails(source);
+
+    this.chartName = chartName;
+    this.chartUrl = chartUrl;
+    this.chartVersion = chartVersion;
+    this.tmpDir = pullHelmRepo(chartUrl, chartName, chartVersion);
+
+    const chartYamlFilePath = path.join(this.tmpDir, this.chartName, CHART_YAML);
+    const contents = Yaml.load(chartYamlFilePath);
+
+    if (contents && contents.length === 1 && contents[0].dependencies) {
```

Comment:
`contents` is never undefined.

```suggestion
    if (contents.length === 1 && contents[0].dependencies) {
```

Also I don't understand the `[0]`...what happens if `contents.length > 1`? what does this mean? can you paste an example `Chart.yaml` with dependencies?

Reply:
Ohh, the `contents[0]` is because `Yaml.load` returns a list,
```
(method) Yaml.load(urlOrFile: string): any[]

Downloads a set of YAML documents (k8s manifest for example) from a URL or a file and returns them as javascript objects.

Empty documents are filtered out.

@param urlOrFile — a URL of a file path to load from

@returns — an array of objects, each represents a document inside the YAML
```

You are correct, it would never be undefined. Will update it. 

`contents.length > 1` should not happen since helm chart should just have one `Chart.yaml`. I can switch to reading and parsing the Yaml file instead of Yaml.load. Let me know if that would be more readable.

Reply:
Here is a Chart.yaml with dependencies: https://github.com/bitnami/charts/blob/main/bitnami/airflow/Chart.yaml#L22-L35